lib/reply.js

```diff
@@ -139,9 +139,15 @@ function onSendEnd (reply, payload) {
     flatstr(payload)
   }
 
+  if (typeof payload !== 'string' && !Buffer.isBuffer(payload)) {
+    handleError(reply, new Error(`Serializer for Content-Type '${contentType}' can not handle payload of type '${typeof payload}'`))
+    return
+  }
```

Comment:
I think we are already doing a typeof earlier, can you put that into a variable and reuse?

Also, I think this might go as an else branch of the previous if.

Reply:
Not possible because `serializer` can produce a different type.